pramen-py/src/pramen_py/metastore/reader.py

```diff
@@ -57,8 +58,96 @@ def _apply_uppercase_to_columns_names(
         else:
             return df
 
+    def _extract_dates_from_file_name(
+        self, path: str, target_partition_name: str
+    ) -> Optional[str]:
+        def is_file_hidden(column_name_: str, date_: str) -> bool:
+            if len(column_name_) > 0 and column_name_[0] == "_":
```

Comment:
Maybe this function could be simplified to be more readable using `column_name_.startswith("_")`